src/components/quad-status/BatteryIcon.vue

```diff
@@ -0,0 +1,136 @@
+<template>
+  <div class="battery-icon">
+    <div class="quad-status-contents">
+      <div
+        class="battery-status"
+        :class="classes"
+        :style="{ width: batteryWidth + '%' }"
+      />
+    </div>
+  </div>
+</template>
+<script>
+const NO_BATTERY_VOLTAGE_MAXIMUM = 1.8; // Maybe is better to add a call to MSP_BATTERY_STATE but is not available for all versions
+
+export default {
+  props: {
+    voltage: {
+      type: Number,
+      default: 0,
+    },
+    vbatmincellvoltage: {
+      type: Number,
+      default: 1,
+    },
+    vbatmaxcellvoltage: {
+      type: Number,
+      default: 1,
+    },
+    vbatwarningcellvoltage: {
+      type: Number,
+      default: 1,
+    },
+  },
+  computed: {
+    nbCells() {
+      let nbCells = Math.floor(this.voltage / this.vbatmaxcellvoltage) + 1;
+      if (this.voltage === 0) {
+        nbCells = 1;
+      }
+      return nbCells;
+    },
+    min() {
+      return this.vbatmincellvoltage * this.nbCells;
+    },
+    max() {
+      return this.vbatmaxcellvoltage * this.nbCells;
+    },
+    warn() {
+      return this.vbatwarningcellvoltage * this.nbCells;
+    },
+    isEmpty() {
+      return (
+        this.voltage < this.min && this.voltage > NO_BATTERY_VOLTAGE_MAXIMUM
+      );
+    },
+    classes() {
+      if(this.batteryState) {
+        return {
+          "state-ok": this.batteryState === 0,
+          "state-warning": this.batteryState === 1,
+          "state-empty": this.batteryState === 2,
+          // TODO: BATTERY_NOT_PRESENT
+          // TODO: BATTERY_INIT
+        };
```

Comment:
If someone has firmware ready with `batteryState` would be good to check.

Reply:
Can check in a few hours

Reply:
Working now. Tested with and without battery. Icon is live.